### ISSUE TITLE
Duplicate WRN_ConvertingNullableToNonNullable warnings are reported for some scenarios

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29960]
```

        [Fact]
        public void MultipleConversions_Explicit_04()
        {
            var source =
@"struct S
{
    public static explicit operator A?(S s) => throw null;
}
class A
{
    internal void F() { }
}
class B : A
{
}
class C
{
    static void F(S s)
    {
        var b1 = (B)s;
        b1.F();
        B b2 = (B)s;
        b2.F();
        A a = (B)s;
        a.F();
    }
}";
            var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition });
            // PROTOTYPE(NullableReferenceTypes): Should only report one WRN_ConvertingNullableToNonNullable
            // warning for `B b2 = (B)s;` and `A a = (B)s;`.
            comp.VerifyDiagnostics(
                // (16,18): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         var b1 = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(16, 18),
                // (17,9): warning CS8602: Possible dereference of a null reference.
                //         b1.F();
                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "b1").WithLocation(17, 9),
                // (18,16): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         B b2 = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(18, 16),
                // (18,16): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         B b2 = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(18, 16),
                // (19,9): warning CS8602: Possible dereference of a null reference.
                //         b2.F();
                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "b2").WithLocation(19, 9),
                // (20,15): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         A a = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(20, 15),
                // (20,15): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         A a = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(20, 15),
                // (21,9): warning CS8602: Possible dereference of a null reference.
                //         a.F();
                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "a").WithLocation(21, 9));
        }
```

Duplicates:
```
                // (18,16): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         B b2 = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(18, 16),
                // (18,16): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         B b2 = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(18, 16),
                // (20,15): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         A a = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(20, 15),
                // (20,15): warning CS8600: Converting null literal or possible null value to non-nullable type.
                //         A a = (B)s;
                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(20, 15),
```
